server/src/main/java/org/elasticsearch/index/mapper/Uid.java

```diff
@@ -112,7 +112,7 @@ private static BytesRef encodeBase64Id(String id) {
     }
 
     private static BytesRef encodeUtf8Id(String id) {
-        byte[] b = new byte[1 + UnicodeUtil.maxUTF8Length(id.length())];
+        byte[] b = new byte[1 + UnicodeUtil.calcUTF16toUTF8Length(id, 0, id.length())];
```

Comment:
This needed to be correctly sized now otherwise we trip an assertion in the version map. That's probably a good idea anyway, looping the id twice now but saving zeroing out 3x as many bytes as needed is probably a wash performance wise.